### DISCUSSION TITLE
automatic migration ways other than using prisma

### DISCUSSION THREAD

**Discussion by bkyerv** [planetscale/discussion#310] [FAQs]
Is there any way to do an automatic migration other than using prisma or cloud services? if not, what would be the best way to do it? Is it possible to write all queries in a separate file with sql extension and pass it to `conn.execute` method so to run all queries when server is spinned in the beginning?

**Comment by orware** [comment]
Hi @bkyerv,

What kind of programming language would you expect to be using? You mentioned Prisma, but I don't know if that automatically implies that you would be creating a Node.js project.

The last part of your response isn't that much different from how most migration systems actually work, so it is possible to create your own simplified automatic migration process should you wish to, but that would mostly depend on your own wishes for your project.

The main thing to keep in mind with a PlanetScale database is that your `main` branch is typically going to be set as a production branch, which won't allow any direct DDL/schema changes on it, so you can't directly run your migrations against that `main` branch in this situation.

Instead, you'll want to be connected to a separate development branch, which would allow direct DDL/schema operations to occur on it, and make those changes there.

Once the schema changes have been made on your development branch, you would take advantage of our [deploy request](https://planetscale.com/docs/concepts/deploy-requests) process to get your schema changes merged back into your `main` branch.
